docs/modules/ROOT/pages/high_level/rbac_design.adoc

```diff
@@ -0,0 +1,227 @@
+
+== Access Type
+Alcor rbac supports a few of access type
+
+. admin_only -- Only admin role can access
+. rule_any -- Every one can access
+. admin_or_owner -- Admin role or resource owner can access
+. multi_roles -- Only specified roles can access
+. multi_roles_or_owner -- Only specified roles or admin can access
+
+== Permission Granularity
+Alcor can support method and resource field check granularity.
+
+. Method -- The method granularity determines whether the user role can access the method.
+. Resource field -- The field granularity determines whether a user role can be accessed, modified, deleted, or added
+
+== WorkFlow
+Rbac check around a http request.
+
+. a request come
+. rbac check method access, failed throw exception
+. rbac check field access, failed throw exception
+. handle request
+. rbac check field access, failed hide the field
+. response the request
+
+== rbac_policy.json example
+....
+{
+  "serviceRules": [
+    {
+      "name": "create_vpc",
+      "method_type": "POST",
+      "description": "create a vpc",
+      "rule_type": "rule_any",
+      "roles": [],
+      "field_rbac_rules": [
+        {
+          "name": "shared",
+          "description": "create a shared vpc",
+          "rule_type": "admin_only",
+          "roles": ["admin"]
+        },
+        {
+          "name": "router:external",
+          "description": "create an external vpc",
+          "rule_type": "admin_only",
+          "roles": ["admin"]
+        },
+        {
+          "name": "is_default",
+          "description": "Specify ''is_default'' attribute when creating a vpc",
+          "rule_type": "admin_only",
+          "roles": ["admin"]
+        },
+        {
+          "name": "port_security_enabled",
+          "description": "Specify ''port_security_enabled'' attribute when creating a vpc",
+          "rule_type": "rule_any",
+          "roles": []
+        },
+        {
+          "name": "segments",
+          "description": "Specify ''segments'' attribute when creating a vpc",
+          "rule_type": "admin_only",
+          "roles": ["admin"]
+        },
+        {
+          "name": "provider:network_type",
+          "description": "Specify ''provider:network_type'' attribute when creating a vpc",
+          "rule_type": "admin_only",
+          "roles": ["admin"]
+        },
+        {
+          "name": "provider:physical_network",
+          "description": "Specify ''provider:physical_network'' attribute when creating a vpc",
+          "rule_type": "admin_only",
+          "roles": ["admin"]
+        },
+        {
+          "name": "provider:segmentation_id",
+          "description": "Specify ''provider:segmentation_id'' attribute when creating a vpc",
+          "rule_type": "admin_only",
+          "roles": ["admin"]
+        }
+      ]
+    },
+    {
+      "name": "get_vpc",
+      "method_type": "GET",
+      "description": "Get or list vpc",
+      "rule_type": "admin_or_owner",
```

Comment:
@Gzure if rule_type=admin_or_owner, why is the value of roles required to be admin? What would be the roles of the token included in our X-Token-info header field?

Reply:
> @Gzure if rule_type=admin_or_owner, why is the value of roles required to be admin? What would be the roles of the token included in our X-Token-info header field?
Admin_or_owner rule type will ignore the roles.

Reply:
Okay in this case, please keep the example as accurate as possible. maybe remove any entry in the roles array. @Gzure

Reply:
> Okay in this case, please keep the example as accurate as possible. maybe remove any entry in the roles array. @Gzure

ok

Reply:
> Okay in this case, please keep the example as accurate as possible. maybe remove any entry in the roles array. @Gzure

Okay, I will remove the roles.